src/v3/src/transformer/phone/phoneEnrollmentTransformer.ts

```diff
@@ -76,37 +77,62 @@ export const transformPhoneEnrollment: IdxStepTransformer = ({ formBag, transact
     },
   };
 
-  const stepper: StepperLayout = {
-    type: UISchemaLayoutType.STEPPER,
-    elements: [
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          smsInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          smsStepSubmitButton,
-        ],
-      },
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          voiceInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          voiceStepSubmitButton,
-        ],
-      },
-    ],
+  const phoneOptionLayouts: { [key:string]: UISchemaLayout | StepperLayout } = {
+    sms: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        smsInfoTextElement,
+        phoneNumberElement,
+        smsStepSubmitButton,
+      ],
+    },
+    voice: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        voiceInfoTextElement,
+        phoneNumberElement,
+        voiceStepSubmitButton,
+      ],
+    },
+    both: {
+      type: UISchemaLayoutType.STEPPER,
+      elements: [
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            smsInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            smsStepSubmitButton,
+          ],
+        },
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            voiceInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            voiceStepSubmitButton,
+          ],
+        },
+      ],
+    } as StepperLayout,
   };
 
   uischema.elements = [
     titleElement,
-    stepper,
   ];
 
+  const phoneMethodOptions = methodTypeElement!.options!.inputMeta.options!;
+  if (phoneMethodOptions.length === 1) {
+    const methodType = phoneMethodOptions[0].value.toString();
+    uischema.elements.push(phoneOptionLayouts[methodType]);
```

Comment:
In addition to this, you should also append the `actionParams` to the applicable button, setting the `authenticator.methodType` as the first methodOption value.

Reply:
@glenfannin-okta do the `actionParams` need to be set in the case where both phone options are present?

Reply:
Nope, when both are present no action params should be added (since its controlled by the radio button selection)